crates/flipperzero/src/furi/message_queue.rs

```diff
@@ -55,11 +55,7 @@ impl<M: Sized> MessageQueue<M> {
             .into()
         };
 
-        if status.is_ok() {
-            Ok(unsafe { out.assume_init() })
-        } else {
-            Err(status)
-        }
+        status.into_result().map(|_| unsafe { out.assume_init() })
```

Comment:
IMO, this is the case where `match` looks better (given `_` binding and `unsafe`), but this may just be a style preference

Reply:
So like the following?

```rust
match status.into_result() {
    Err(err) => Err(err),
    Ok(_) => Ok(out.assume_init()),
}
```

I guess using an `if let` expression would also work here:

```rust
if let Err(err) = status.into_result() {
    Err(err)
} else {
    unsafe { out.assume_init() }
}
```

or even:

```rust
let _ = status.into_result()?;

unsafe { out.assume_init() }
```

Reply:
I think that last one is the most explicit that we're not using the actual status code for anything.

I updated the code to match that style.